chart/templates/webserver/webserver-ingress.yaml

```diff
@@ -99,8 +114,12 @@ spec:
               servicePort: {{ .servicePort }}
               {{- end }}
           {{- end }}
-      {{- if . }}
-      host: {{ . | quote }}
+      {{- $hostname := . -}}
+      {{- if and $.Values.ingress.web.hosts ($.Values.ingress.web.hosts | first | kindIs "string" | not) }}
+      {{- $hostname = .name -}}
+      {{- end }}
+      {{- if $hostname }}
+      host: {{ $hostname | quote }}
```

Comment:
Fair warning, untested, but this seems cleaner to me. (Checking `.` instead of the first element)